core/src/main/java/jenkins/security/UpdateSiteWarningsMonitor.java

```diff
@@ -0,0 +1,176 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.security;
+
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.PluginWrapper;
+import hudson.model.AdministrativeMonitor;
+import hudson.model.UpdateSite;
+import jenkins.model.Jenkins;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+/**
+ * Administrative monitor showing plugin/core warnings published by the configured update site to the user.
+ *
+ * Terminology overview:
+ * - Applicable warnings are those relevant to currently installed components
+ * - Active warnings are those actually shown to users.
+ * - Hidden warnings are those _not_ shown to users due to them being configured to be hidden.
+ * - Inapplicable warnings are those that are not applicable.
+ *
+ * The following sets may be non-empty:
+ * - Intersection of applicable and active
+ * - Intersection of applicable and hidden
+ * - Intersection of hidden and inapplicable (although not really relevant)
+ * - Intersection of inapplicable and neither hidden nor active
+ *
+ * The following sets must necessarily be empty:
+ * - Intersection of applicable and inapplicable
+ * - Intersection of active and hidden
+ * - Intersection of active and inapplicable
+ *
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class UpdateSiteWarningsMonitor extends AdministrativeMonitor {
+    @Override
+    public boolean isActivated() {
+        return !getActiveCoreWarnings().isEmpty() || !getActivePluginWarningsByPlugin().isEmpty();
```

Comment:
Well, don't forget we render UI every time the administrator navigates to another page or if the page gets reloaded. It may be a premature before you finish the implementation, but IMHO it deserves fixing before merging.